tests/integration/policy/envoy_ratelimit_test.go

```diff
@@ -0,0 +1,196 @@
+//  Copyright 2020 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package policy
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"fortio.org/fortio/fhttp"
+	"fortio.org/fortio/periodic"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/bookinfo"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/istio/ingress"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/kube"
+	"istio.io/istio/pkg/test/scopes"
+)
+
+var (
+	ist         istio.Instance
+	bookinfoNs  namespace.Instance
+	ratelimitNs namespace.Instance
+	ing         ingress.Instance
+)
+
+func TestRateLimiting_DefaultLessThanOverride(t *testing.T) {
+	framework.
+		NewTest(t).
+		Run(func(ctx framework.TestContext) {
+			res := sendTraffic(ing, t, "Sending traffic...", 300)
+			got429s := float64(res.RetCodes[http.StatusTooManyRequests])
+
+			if got429s <= 0 {
+				t.Errorf("Bad metric value for rate-limited requests (429s): got %f, want at least 1", got429s)
+			}
+		})
+}
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite(m).
+		RequireSingleCluster().
+		Label(label.CustomSetup).
+		Setup(istio.Setup(&ist, nil)).
+		Setup(testSetup).
+		Run()
+}
+
+func testSetup(ctx resource.Context) (err error) {
+	bookinfoNs, err = namespace.New(ctx, namespace.Config{
```

Comment:
Is there any reason not using echo app? It is the default app for testing.

Reply:
This was written to provide comparison with old tests but changing to use echo